lib/fibonacci.rb

```diff
@@ -1,8 +1,22 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
-# Hint, you may want a recursive helper method
+# Time Complexity - O(1)
+# Space Complexity - O(n)
+
 def fibonacci(n)
```

Comment:
This is definitely not time complexity O(1) since the amount of time it takes changes as n gets bigger.  However both would be O(n)